### DISCUSSION TITLE
Can I use my phone as a tracker?

### DISCUSSION THREAD

**Discussion by Bigbrian5** [eusonlito/GPS-Tracker#42] [Q&A]
@eusonlito I see that you are using a aftermarket GPS tracker, but I was wondering like in traccar if I could use my phone as the GPS tracker?

**Comment by eusonlito** [comment]
Yes, you can use the traccar application itself as a tracker, since it uses the OsmAnd protocol. Simply configure an OsmAnd port in the platform and start receiving requests.

You can also use the application https://github.com/mendhak/gpslogger and send every position to url:

```
http://SERVER:PORT/?lat=%LAT&lon=%LON&timestamp=%TIMESTAMP&speed=%SPD&direction=%DIR&id=%SER&maker=gpslogger&hdop=1
```

**Reply by Bigbrian5** [reply]
Okay great! Thanks for the info, this is much better than Traccar for tracking motorcycle rides!

**Reply by Bigbrian5** [reply]
Now that I have it setup where do I enter this command?
![image](https://github.com/eusonlito/GPS-Tracker/assets/122059465/12ee9b61-9f61-456c-92e8-9a6fd03d6289)

**Reply by eusonlito** [reply]
This command can only be sent by SMS to device. To send more commands `Devices` > `List` > `Messages`.

**Reply by Bigbrian5** [reply]
So I have the osmand protocol working yet the data isn't coming through?
![image](https://github.com/eusonlito/GPS-Tracker/assets/122059465/a150441b-1a7d-43d8-b6db-03836eefcf1e)

**Reply by eusonlito** [reply]
Request is GET and check query names (`lat`, `lon`, etc...):

```
http://SERVER:PORT/?lat=%LAT&lon=%LON&timestamp=%TIMESTAMP&speed=%SPD&direction=%DIR&id=%SER&maker=gpslogger&hdop=1
```

**Reply by eusonlito** [reply]
`id` is the device ID (serial), `bearing` should be `direction`? `accuracy` should be `hdop`?

**Reply by Bigbrian5** [reply]
Where Do I place that GET request? In Servers>Status>Parser? Yeah I do think bearing is equal to direction and accuracy is hdop.

**Reply by eusonlito** [reply]
You should configure the gpslogger `Menu` > `Custom URL`:

*  `Log to custom URL`: true
*  `Discard offline locations`: true
* `HTTP Method`: `GET`
* `URL`

```
http://(here your server domain):(here your OsmAnd port)/?lat=%LAT&lon=%LON&timestamp=%TIMESTAMP&speed=%SPD&direction=%DIR&id=%SER&maker=gpslogger&hdop=1
```

**Reply by Bigbrian5** [reply]
I have an iphone and I am trying to use the traccar client app on my cell phone, I looked at the other thread on your github but it seemed to not come to a conclusion?

**Reply by eusonlito** [reply]
Ok, let me work to adapt traccar to OsmAnd.

**Reply by Bigbrian5** [reply]
That would be great!

**Reply by eusonlito** [reply]
@Bigbrian5 done, protocol updated.

**Reply by Bigbrian5** [reply]
So what did you do exactly? I did a git pull and a ./docker/build.sh but it seems that nothing happened and its not pulling information from that post command
[2024-02-29T14:50:53+00:00] POST /8091?id=f4ba6d86-0de1-4001-b5e7-a0a2243db8d5×tamp=1709218252&lat=42.578130&lon=-71.195603&speed=0&bearing=202.9931586365065&altitude=31.54787266068161&accuracy=4.744020813364718&batt=20&charge=true HTTP/1.1
Host: gpst.largebrain.org
User-Agent: TraccarClient/59 CFNetwork/1492.0.1 Darwin/23.3.0
Content-Length: 0
Accept: */*
Accept-Encoding: gzip
Accept-Language: en-US,en;q=0.9
Cdn-Loop: cloudflare

**Reply by eusonlito** [reply]
mmm what is `/8091`? your request should be:

```
POST /?id=f4ba6d86-0de1-4001-b5e7-a0a2243db8d5×tamp=1709218252&lat=42.578130&lon=-71.195603&speed=0&bearing=202.9931586365065&altitude=31.54787266068161&accuracy=4.744020813364718&batt=20&charge=true HTTP/1.1
```

**Reply by eusonlito** [reply]
For example:

![image](https://github.com/eusonlito/GPS-Tracker/assets/644551/31aa40fe-962d-4920-bdf1-795ee2c7ed5c)

- **Device Identifier**: Same as your GPS Tracker serial.
- **Server URL**: http://gpst.largebrain.org:8091

**Reply by Bigbrian5** [reply]
I have a path prefix from traefik which tacs on "/8091" to the  htpps://gpst.largebrain.org because when I try to attach 8091 which is the normal Osmand port to the traefik service then the service fails so I was attaching it to a path prefix unless there is something else I can change in my docker compose? 
![image](https://github.com/eusonlito/GPS-Tracker/assets/122059465/269997c2-f7e5-476d-b383-df3a94f8e02a)

**Reply by eusonlito** [reply]
You should connect to a port, not to a path. The port is 8091 and the path should be empty.

**Reply by Bigbrian5** [reply]
I guess I'm a little lost on this whole front haha. So I will start from the top!
Do I need port 8091?
If so should I be able to tack it on to the end of https://gpst.largebrain.org:8091? 
If I can't do that the only way that I have found to pass information to that OsmAnd protocol port is by adding the /8091 prefix on the end

**Reply by eusonlito** [reply]
You can use any port, simply configure the OsmAnd port on the platform with the port of your choice. Logically in traccar app you have to indicate the host and port in the connection URL through http://HOST:PORT.

Maybe the rule defined in traefik should be simply by port, instead of by host and URL, no?

**Reply by eusonlito** [reply]
maybe adding a `.loadbalancer.server.port=8091`?

**Reply by eusonlito** [reply]
maybe the current `.loadbalancer.server.port=8091` should be defined in `gpstracker-rtr-8091` service?

**Reply by Bigbrian5** [reply]
To my knowledge you can only have one loadbalancer server port per each service that's why I added the prefix and the other service becasue if I have 2 ports on the same service it doesn't work

**Reply by eusonlito** [reply]
But, why do you have 2 different services to 8091 port?

**Reply by Bigbrian5** [reply]
you just get a 502 bad gateway error with 2 ports, I commented out the 8091:8091 to test and it still doesn't work

**Reply by Bigbrian5** [reply]
Can't I just modify the Location.php to take in account the prefix of /8091 ??
    protected function bodyIsValidRegExp(): string
    {
        return '/(GET|POST) \/\?(.*) HTTP\/1/';
    }

    /**
     * @return void
     */

**Reply by eusonlito** [reply]
```
return '/(GET|POST) \/[^\?]*\?(.*) HTTP\/1/';
```

**Reply by eusonlito** [reply]
Updated :)

**Reply by Bigbrian5** [reply]
So after the update the gpstracker-app won't worked and its caused by a few PHP errors in two php files one in the Connection.php line 829 and another in  the Connector.php f line 65. As pre requisites I did login to the mysql database and make sure the user gpstracker has full grant privileges on the gpstracker database so I dunno what changed?
[gpstracker-app-2024-03-01T00-22-32.log](https://github.com/eusonlito/GPS-Tracker/files/14454873/gpstracker-app-2024-03-01T00-22-32.log)

**Reply by eusonlito** [reply]
Have you deployed with my scripts? Check the `DB_` config values in the `.env` file inside the project path.

**Reply by Bigbrian5** [reply]
Yes I have deployed with your scripts in a separate docker compose and I still get that error

**Reply by Bigbrian5** [reply]
which leads to a 500 system error on the main app website

**Reply by eusonlito** [reply]
Yes, because the database configuration is not set correctly. Do you have checked the `DB_` `.env` values?

**Reply by Bigbrian5** [reply]
Here is how everything is set with the `DB_` `.env` values, it looks exactly like what you have from your github.
![image](https://github.com/eusonlito/GPS-Tracker/assets/122059465/4628a704-15e4-4268-b4b1-60b1ec50b98c)

**Reply by Bigbrian5** [reply]
Is there a way to configure this database manually since mysql is sad for whatever reason?

**Comment by eusonlito** [comment]
The error is: `Access denied for user 'gpstracker'@'172.18.0.4'` and you have configured `DB_HOST` with `gpstracker-mysql`. Do you have changed any value on docker-compose?

**Reply by Bigbrian5** [reply]
The only changes I have made to the docker-compose is to add my traefik labels everything else is stock

**Reply by eusonlito** [reply]
can you connect from inside the container to mysql?

**Reply by Bigbrian5** [reply]
Yes
![image](https://github.com/eusonlito/GPS-Tracker/assets/122059465/220224e8-1e56-499b-8e8f-28170e3dfec0)

**Reply by Bigbrian5** [reply]
![image](https://github.com/eusonlito/GPS-Tracker/assets/122059465/4bc1511f-b8f3-413b-9783-b12dfc9b4113)

**Reply by eusonlito** [reply]
No, from gpstracker source container to gpstracker mysql container.

**Reply by Bigbrian5** [reply]
I can't gives me an Error 2002
![image](https://github.com/eusonlito/GPS-Tracker/assets/122059465/1131607d-74f7-4cbf-bb2f-5e5e426d83b7)

**Reply by eusonlito** [reply]
You need to connect as is configured in `.env` file, with `-h gpstracker-mysql`.

Can you do a `cat .env`?

**Reply by Bigbrian5** [reply]
It was the gpstracker-network not existing, dunno how that happened ¯\_(ツ)_/¯
